### PR TITLE
[openvpn] Add pmacct json-log support

### PR DESCRIPTION
**Pull request by trublast**
Closed in favor of https://github.com/deckhouse/deckhouse/pull/3686.
## Description
Add `pmacct` JSON-log support for OpenVPN.

## Why do we need it, and what problem does it solve?
Sometimes it's required to check user actions over VPN.

## What is the expected result?
Add sidecar with `pmacctd`.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: openvpn
type: feature
summary: Add `pmacct` JSON-log support for OpenVPN.
```

### CODE REVIEW

File: modules/500-openvpn/openapi/doc-ru-config-values.yaml [hunk 1]
```diff
@@ -131,6 +131,12 @@ properties:
       Структура, аналогичная  `spec.tolerations` в Kubernetes Pod.
 
       Если ничего не указано или указано `false` — будет [использоваться автоматика](https://deckhouse.ru/documentation/v1/#выделение-узлов-под-определенный-вид-нагрузки).
+  pmacctEnabled:
+    description: |
+      Включить логирование IP соединений в виде json-логов для клиентских openvpn подключений.
```
Comment (apolovov) [inline modules/500-openvpn/openapi/doc-ru-config-values.yaml:136]: Mind the foreign-word-based-words, there must be dash.
```suggestion
      Включить логирование IP-соединений в виде json-логов для клиентских openvpn-подключений.
```

**Review by apolovov — CHANGES_REQUESTED** [review]
It is important to explain the feature in docs/README.md.
I think, there must be additional header like "User's traffic audit" with the explanation.

**Comment by apolovov** [comment]
Also:
* fix the DCO (add a string `Signed-off-by: John Doe <john.doe@example.com>` and repush the commit).
* fix the description — no one knows what pmacct is. Describe the feature a bit wider.
* I prepared the Changelog entries — pls, add the description here.

File: modules/500-openvpn/docs/README.md [hunk 2]
```diff
@@ -26,3 +26,9 @@ The following parameters are pushed to the user's computer after connecting to t
 - a route to the local network;
 - a route to the cluster service network;
 - a route to the Pod network.
+
+## User's traffic audit
+
+The module has the ability to enable logging of all user connections made via VPN in json format. Traffic is grouped
+by `src_ip, dst_ip, src_port, dst_port, ip_proto` fields. Container logs can be collected and sent for storage or analysis
+using the [log-shipper](../460-log-shipper/) module.
```
Comment (z9r5) [inline modules/500-openvpn/docs/README.md:34]:
```suggestion
The module has the ability to enable logging of all user connections made via VPN in JSON format. Traffic is grouped
by `src_ip, dst_ip, src_port, dst_port, ip_proto` fields. Container logs can be collected and sent for storage or analysis
using the [log-shipper](../460-log-shipper/) module.
```

File: modules/500-openvpn/docs/README_RU.md [hunk 3]
```diff
@@ -26,3 +26,9 @@ Web-интерфейс проинтегрирован с модулем [user-au
 - маршрут в локальную сеть;
 - маршрут в сервисную сеть кластера;
 - маршрут в сеть Pod'ов.
+
+## Аудит пользовательских соединений
+
+Модуль имеет возможность включить логирование всех пользовательских соединений, выполненных через VPN, в json-формате. Группировка трафика
+происходит по полям `src_ip, dst_ip, src_port, dst_port, ip_proto`. Логи контейнера могут быть собраны и отправлены на хранение или анализ
+с помощью модуля [log-shipper](../460-log-shipper/).
```
Comment (z9r5) [inline modules/500-openvpn/docs/README_RU.md:34]:
```suggestion
Модуль позволяет включить логирование всех пользовательских VPN-соединений в JSON-формате. Группировка трафика происходит по полям `src_ip, dst_ip, src_port, dst_port, ip_proto`. Логи контейнера могут быть собраны и отправлены на хранение или анализ с помощью модуля [log-shipper](../460-log-shipper/).
```

File: modules/500-openvpn/openapi/doc-ru-config-values.yaml [hunk 4]
```diff
@@ -131,6 +131,15 @@ properties:
       Структура, аналогичная  `spec.tolerations` в Kubernetes Pod.
 
       Если ничего не указано или указано `false` — будет [использоваться автоматика](https://deckhouse.ru/documentation/v1/#выделение-узлов-под-определенный-вид-нагрузки).
+  pmacctEnabled:
+    description: |
+      Включить логирование IP-соединений в виде json-логов для клиентских openvpn-подключений.
+      
+      Все соединения внутри `tun` интерфейсов будут собраны через `libpcap` и записаны в лог
+      в виде `{"event_type": "purge", "ip_src": "172.25.175.10", "ip_dst": "10.222.0.10", "port_src": 32172, "port_dst": 53, "ip_proto": "udp", "packets": 1, "bytes": 53}`
```
Comment (z9r5) [inline modules/500-openvpn/openapi/doc-ru-config-values.yaml:139]:
```suggestion
      Включить логирование VPN-подключений в JSON-формате.
      
      Все соединения внутри `tun` интерфейсов будут собраны через `libpcap` и записаны в лог
      в виде `{"event_type": "purge", "ip_src": "172.25.175.10", "ip_dst": "10.222.0.10", "port_src": 32172, "port_dst": 53, "ip_proto": "udp", "packets": 1, "bytes": 53}`.
```

File: modules/500-openvpn/openapi/config-values.yaml [hunk 5]
```diff
@@ -225,6 +225,15 @@ properties:
     type: boolean
     default: false
     x-examples: [true, false]
+  pmacctEnabled:
+    description: |
+      Enable json-log IP traffic accounting for openvpn client connections.
```
Comment (z9r5) [inline modules/500-openvpn/openapi/config-values.yaml:230]:
```suggestion
      Enable logging of VPN connections in JSON format.
```

**Comment by z9r5** [comment]
Some tests didn't pass...

**Comment by apolovov** [comment]
https://github.com/deckhouse/deckhouse/pull/3686

**Pull request closed without merging**